shared/teams/role-picker/container.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import {connect} from 'react-redux'
+import * as I from 'immutable'
+import {compose, withState} from 'recompose'
+import RolePicker from '.'
+import * as Creators from '../../actions/teams/creators'
+import * as Constants from '../../constants/teams'
+
+import type {TypedState} from '../../constants/reducer'
+
+type StateProps = {
+  _memberInfo: I.Set<Constants.MemberInfo>,
+  you: ?string,
+  username: string,
+  teamname: string,
+}
+
+const mapStateToProps = (state: TypedState, {routeProps}): StateProps => ({
+  _memberInfo: state.entities.getIn(['teams', 'teamNameToMembers', routeProps.get('teamname')], I.Set()),
+  teamname: routeProps.get('teamname'),
+  username: routeProps.get('username'),
+  you: state.config.username,
+})
+
+export type Role = null | 'reader' | 'writer' | 'admin' | 'owner'
```

Comment:
ah, you have this here already...should make non-nullable and have only one type definition